.github/actions/backup-database/action.yml

```diff
@@ -0,0 +1,44 @@
+name: Backup DB
+description: Backup DB and upload GitHub artefact
+
+inputs:
+  environment:
+    description: The name of the environment
+    required: true
+  azure-credentials:
+    description: Azure credentials
+    required: true
+
+runs:
+  using: composite
+
+  steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Login
+      uses: azure/login@v1
+      with:
+        creds: ${{ inputs.azure-credentials }}
+
+    - name: Set AKS credentials
+      shell: bash
+      run: az aks get-credentials -g s189p01-tsc-pd-rg -n s189p01-tsc-production-aks
+
+    - name: Install kubectl
+      uses: azure/setup-kubectl@v3
+
+    - name: Install konduit
+      shell: bash
+      run: make install-konduit
+
+    - name: Backup database
+      shell: bash
+      run: |
+        bin/konduit.sh cpd-ecf-${{ inputs.environment }}-web -- pg_dump -E utf8 --compress=1 --clean --if-exists --no-privileges --no-owner --verbose -f backup-${{ inputs.environment }}.sql.gz
```

Comment:
Should we be concerned about storing production data outside of Azure/in GitHub artifacts, albeit temporarily?

Reply:
Could we combine backup and restore in the same job? So we don't need to store anything

Reply:
Or siimply store it in azure storage, such as: https://github.com/DFE-Digital/register-trainee-teachers/blob/main/.github/workflows/database-backup.yml

Reply:
amended now and combined under one job, if you can both have another look, thanks

Reply:
Looks good to me 👍